### ISSUE TITLE
Issues with table configuration after using 'read In' Functions for ref/alt/coverage tables 

### ISSUE THREAD

**Issue by SawyerRSmith** [bailey-lab/miplicorn#27]
Hi!
the read in function appears to be mixing up which columns are which for alt/cov/ref. Right now my alt and coverage columns have their values switched. I checked with the table outputs directly from MIPTools to be sure. 

Here is my code block.
ref<-"~/R/extended-haplotype-Uganda/data/hap3files/reference_AN_table_IBC.csv"
cov<-"~/R/extended-haplotype-Uganda/data/hap3files/coverage_AN_table_IBC.csv"
alt<-"~/R/extended-haplotype-Uganda/data/hap3files/alternate_AN_table_IBC.csv"
data<-miplicorn::read_tbl_ref_alt_cov(ref, cov, alt, targeted == "Yes")

I am wondering if the values for ref/alt/cov need to be in any specific order and if this is documented?

Thanks so much!

**Comment by JeffAndBailey** [comment]
Is there away to check to ensure no one screws this up?

**Comment by arisp99** [comment]
Not really, the three files have the exact same structure so it is very difficult to infer which file is which if the user inputs them in random order. 

From a user point of view, one thing that might be helpful is to write the function call as I did earlier and include the argument name in the call. This makes it explicit which table the function is expecting and in what order.

**Comment by JeffAndBailey** [comment]
They do have teltale signs or we should add some explict signs in MIPtools.  I have been also thinking whether we make a simple concatenated format   ref:alt:coverage combining the 3 tables as an alternative form someone could use.   Otherwise, we should do some math to double check ...  ref + alt = coverage and   ref > alt in terms of numbers if you look across a good number of sites.

**Comment by arisp99** [comment]
That's a good point. I've been playing around a bit with classes for some of the new functions so I think ultimately that will be the way to go. We can have a class for the combined 3 tables (ref, alt, coverage) which can then check a couple of conditions to ensure we have the right data structure...

re: the concatenated format. I think this is what I am doing now, no? `read_tbl_ref_alt_cov()` reads the three tables at once and combines them together.

``` r
library(miplicorn)

ref_file <- miplicorn_example("reference_AA_table.csv")
alt_file <- miplicorn_example("alternate_AA_table.csv")
cov_file <- miplicorn_example("coverage_AA_table.csv")
read_tbl_ref_alt_cov(
 ref_file,
 alt_file,
 cov_file,
 gene == "atp6" | gene == "crt"
)
#> # A tibble: 832 × 10
#>    sample     gene_id       gene  mutation_name  exonic_func  aa_change targeted
#>    <chr>      <chr>         <chr> <chr>          <chr>        <chr>     <chr>   
#>  1 D10-JJJ-23 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#>  2 D10-JJJ-43 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#>  3 D10-JJJ-55 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#>  4 D10-JJJ-5  PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#>  5 D10-JJJ-47 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#>  6 D10-JJJ-15 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#>  7 D10-JJJ-27 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#>  8 D10-JJJ-10 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#>  9 D10-JJJ-28 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#> 10 D10-JJJ-52 PF3D7_0106300 atp6  atp6-Ala623Glu missense_va… Ala623Glu Yes     
#> # … with 822 more rows, and 3 more variables: ref_umi_count <dbl>,
#> #   alt_umi_count <dbl>, coverage <dbl>
```

<sup>Created on 2021-12-09 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
